pkg/provider/device42/device42.go

```diff
@@ -0,0 +1,175 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package device42
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	kclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	esv1beta1 "github.com/external-secrets/external-secrets/apis/externalsecrets/v1beta1"
+	"github.com/external-secrets/external-secrets/pkg/utils"
+)
+
+const (
+	errNotImplemented                         = "not implemented"
+	errUninitializedProvider                  = "unable to get device42 client"
+	errCredSecretName                         = "credentials are empty"
+	errInvalidClusterStoreMissingSAKNamespace = "invalid clusterStore missing SAK namespace"
+	errFetchSAKSecret                         = "couldn't find secret on cluster: %w"
+	errMissingSAK                             = "missing credentials while setting auth"
+)
+
+type Client interface {
+	GetSecret(secretID string) (D42Password, error)
+}
+
+// Device42 Provider struct with reference to a Device42 client.
+type Device42 struct {
+	client Client
```

Comment:
Is this all the functionality that is needed? There is no way to delete / cleanup a secret?

Reply:
@Skarlso - thank you so much for reviewing this and for all the feedback!!!!
There is no official sdk for device42 and the API is very simple, so I took the approach almost identical to the `passworddepot` implementation which is simple http client based -> https://github.com/external-secrets/external-secrets/pull/2799
I tried to be consistent with this one.
This looked similar enough to the other providers.
It worked as expected when I ran the provider locally - it created a kubernetes secret and sync'ed the password field from device42 into the defined data key.
I just kind of assumed this was all that was needed but if there is something else I should do please let me know.

Reply:
Well, I think it's enough for a very basic provider. If any functionality is needed more I guess it can be added later. :) So this is fine for now.